main.go

```diff
@@ -234,3 +242,28 @@ func webhookServer(port int, withHTTP2 bool) webhook.Server {
 	res := webhook.NewServer(webhookServerOptions)
 	return res
 }
+
+func hasNetworkPolicies(ns string) bool {
+	c, err := client.New(ctrl.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		setupLog.Error(err, "failed to create k8s client for network policy detection")
+		return false
+	}
+
+	policies := &networkingv1.NetworkPolicyList{}
+	if err := c.List(context.Background(), policies, client.InNamespace(ns)); err != nil {
+		setupLog.Error(err, "failed to list network policies", "namespace", ns)
+		return false
+	}
+
+	if len(policies.Items) > 0 {
+		var names []string
+		for _, p := range policies.Items {
+			names = append(names, p.GetName())
+		}
+		setupLog.Info("enabling network policies for operand because networkpolicies exist in the namespace", "namespace", ns, "policies", strings.Join(names, ", "))
```

Comment:
this is not accurate. The purpouse of tis function is to tell the existance of network policies. What we do with the information does nto belong to this function